File: DataModels/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Diplom.DataModels
{
    public class Event
    {
        public int EventId { get; set; }
        public string EventName { get; set; }
        public string Description { get; set; }
        public int? Harmed { get; set; }
        public int? Deaths { get; set; }
        public double? Losses { get; set; }
        public double? Costs { get; set; }
        public DateTime? Date { get; set; }
        public byte[] ImageData { get; set; }

        public EventPosition EventPosition { get; set; }

        public int EmergencyId { get; set; }
        public Emergency Emergency { get; set; }
    }
}

File: ClientApp/src/app/shared/models/event.ts
import { EventPosition } from "./eventPosition";
import { Emergency } from "./emergency";

export class Event{
    public eventId: number;
    public eventName: string;
    public description: string;
    public harmed?: number;
    public deaths?: number;
    public losses?: number;
    public costs?: number;
    public date?: string
    public imageData?: string[];
    public eventPosition: EventPosition;
    public emergency?: Emergency;
    public emergencyId: number;
    
    constructor(){
        this.eventPosition = new EventPosition();
        this.emergency = new Emergency();
        this.harmed = 0;
        this.deaths = 0;
        this.losses = 0;
        this.costs = 0;
    }
}